sos/report/plugins/juju.py

```diff
@@ -53,6 +65,52 @@ def setup(self):
             # logs in the directory.
             self.add_copy_spec("/var/log/juju/*.log")
 
+        # Only run the feature code if this plugin option is set
+        if not self.get_option("is-feature"):
+            return
+
+        juju_user = self.get_option("juju-user")
+        try:
+            pwd.getpwnam(juju_user)
+        except KeyError:
+            # The user doesn't exist, this will skip the rest
+            self._log_warn(f'User "{juju_user}" does not exist, will not collect Juju information.')
+            return
+
+        user_context = f'sudo -u {juju_user}'
+
+        controllers_json = self.collect_cmd_output(f"{user_context} juju controllers --format=json")
+        if controllers_json['status'] == 0:
+            desired_controllers = set(self.get_option('controllers').split(' '))
+            # If a controller option is supplied, use it. Otherwise, get all controllers
+            if desired_controllers and desired_controllers != {''}:
+                controllers = desired_controllers
+            else:
+                controllers = set(json.loads(controllers_json["output"])["controllers"].keys())
+        else:
+            controllers =  {}
+
+        if self.get_option('controllers') and self.get_option('models'):
+            self._log_warn('Options: controllers, models are mutually exclusive.')
+            return
```

Comment:
is it worth moving this test straight after the get_option("juju-user"), this would mean we return any of these to items before actually checking for `juju controllers`?